reframe/core/schedulers/slurm.py

```diff
@@ -193,6 +196,20 @@ def _get_reservation_nodes(self):
         node_descriptions = completed.stdout.splitlines()
         return (SlurmNode(descr) for descr in node_descriptions)
 
+    def _get_excluded_nodes(self):
+        if not self.sched_exclude_nodelist:
+            return set()
+
+        command = 'scontrol show -o -a %s' % self.sched_exclude_nodelist
+        try:
+            completed = os_ext.run_command(command, check=True)
+        except SpawnedProcessError as e:
+            raise JobError('could not retrieve the node description '
+                           'of nodes: %s' % self.sched_exclude_nodelist) from e
+        node_descriptions_output = completed.stdout
+        slurm_nodes = (SlurmNode(descr) for descr in node_descriptions)
```

Comment:
Use directly `completed.stdout` here and please leave a blank line after the `try/except` block.

Reply:
These is also a bug here since I do not split the lines.